supportedBackends/google/pipelines/v2beta/src/main/scala/cromwell/backend/google/pipelines/v2beta/api/request/AbortRequestHandler.scala

```diff
@@ -4,20 +4,21 @@ import akka.actor.ActorRef
 import com.google.api.client.googleapis.batch.BatchRequest
 import com.google.api.client.googleapis.json.GoogleJsonError
 import com.google.api.client.http.HttpHeaders
+import com.typesafe.scalalogging.LazyLogging
 import cromwell.backend.google.pipelines.common.api.PipelinesApiRequestManager._
-import cromwell.backend.google.pipelines.common.api.clients.PipelinesApiAbortClient.{PAPIAbortRequestSuccessful, PAPIOperationAlreadyCancelled, PAPIOperationHasAlreadyFinished}
+import cromwell.backend.google.pipelines.common.api.clients.PipelinesApiAbortClient.{PAPIAbortRequestSuccessful, PAPIOperationHasAlreadyFinished}
 import cromwell.cloudsupport.gcp.auth.GoogleAuthMode
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-trait AbortRequestHandler { this: RequestHandler =>
+trait AbortRequestHandler extends LazyLogging { this: RequestHandler =>
   protected def handleGoogleError(abortQuery: PAPIAbortRequest, pollingManager: ActorRef, e: GoogleJsonError, responseHeaders: HttpHeaders): Try[Unit] = {
-    // No need to fail the request if the job was already cancelled, we're all good
-    if (Option(e.getCode).contains(400) && Option(e.getMessage).contains("Operation has already been canceled")) {
-      abortQuery.requester ! PAPIOperationAlreadyCancelled(abortQuery.jobId.jobId)
-      Success(())
-    } else if (Option(e.getCode).contains(400) && Option(e.getMessage).contains("Operation has already finished")) {
+    // No need to fail the request if the job already terminated, we're all good
+    // As seen in `cromwell.backend.google.pipelines.common.api.clients.PipelinesApiAbortClient` the difference between "finished" and "cancelled" only affects logging
+    // Enhance to be more specific if/when Google implements https://partnerissuetracker.corp.google.com/issues/171993833
```

Comment:
Nice! Thanks for following through on this.

TOL: Is there an additional query-like call that we could make to confirm the reason for the 400, that the job is already terminated?

Reply:
In that ticket the PAPI team said they could make the cancel API idempotent such that cancel requests to an already-stopped job also return success. I believe this accomplishes the goal you have in mind with less change in Cromwell.